tests/sentry/tasks/test_commits.py

```diff
@@ -136,6 +136,41 @@ def test_fetch_error_plugin_error(self, mock_compare_commits):
         assert msg.to == [self.user.email]
         assert "secrets" not in msg.body
 
+    @patch("sentry.plugins.providers.dummy.repository.DummyRepositoryProvider.compare_commits")
+    def test_fetch_error_plugin_error_for_sentry_app(self, mock_compare_commits):
+        org = self.create_organization(owner=self.user, name="baz")
+        sentry_app = self.create_sentry_app(
+            organization=org, published=True, verify_install=False, name="Super Awesome App"
+        )
+
+        repo = Repository.objects.create(name="example", provider="dummy", organization_id=org.id)
+        release = Release.objects.create(organization_id=org.id, version="abcabcabc")
+
+        commit = Commit.objects.create(organization_id=org.id, repository_id=repo.id, key="a" * 40)
+
+        ReleaseHeadCommit.objects.create(
+            organization_id=org.id, repository_id=repo.id, release=release, commit=commit
+        )
+
+        refs = [{"repository": repo.name, "commit": "b" * 40}]
+
+        release2 = Release.objects.create(organization_id=org.id, version="12345678")
```

Comment:
@NisanthanNanthakumar Don't think we need two commits here in this test if we want to keep things simple

Reply:
@scefali There is only one commit. There are 2 releases, but I think that's how get the diff, right?